project/script_IR.tcl

```diff
@@ -1,20 +1,31 @@
-############################################################
-## This file is generated automatically by Vivado HLS.
-## Please DO NOT edit it.
-## Copyright (C) 1986-2017 Xilinx, Inc. All Rights Reserved.
-############################################################
+# Script to generate project for IR
+#   vivado_hls -f script_IR.tcl
+#   vivado_hls -p inputrouter
+# WARNING: this will wipe out the original project by the same name
+
+# create new project (deleting any existing one of same name)
 open_project -reset inputrouter
+
+# source files
+set CFLAGS {-std=c++11 -I../TrackletAlgorithm}
 set_top InputRouter
-add_files ../TrackletAlgorithm/InputRouter.cpp -cflags "-std=c++11"
-add_files -tb ../TestBenches/InputRouter_test.cpp -cflags "-I../TrackletAlgorithm -std=c++11"
-add_files -tb ../emData/IL
-open_solution -reset "solution1"
+add_files ../TrackletAlgorithm/InputRouter.cpp -cflags "$CFLAGS"
+add_files -tb ../TestBenches/InputRouter_test.cpp -cflags "CFLAGS"
+
+# data files
+add_files -tb ../emData/IL/
+
+# solutions
+open_solution "solution1"
+source settings_hls.tcl
 source set_fpga.tcl
-create_clock -period 4 -name default
-#source "./projrouter/solution1/directives.tcl"
-csim_design -compiler gcc
+create_clock -period 250MHz -name default
```

Comment:
Shall we put the clock frequency into one of the global TCL files too to ensure all projects use the same one?

Reply:
As we explore running at higher frequencies, it is possible that one or two of the projects will successfully run at higher frequencies, before all projects achieve this. That is the only argument I see against setting the clock freq. in the common file. I don't mind either way.

Reply:
I can imagine two clock speeds at most, maybe we have a common slow clock and a common fast clock?